JotunnLib/Extensions/GameObjectExtension.cs

```diff
@@ -104,6 +106,174 @@ public static GameObject OrNull(this GameObject @this)
 
             return target;
         }
+
+        /// <summary>
+        ///     Extension method to check if GameObject has a component.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="gameObject"></param>
+        /// <returns></returns>
+        public static bool HasComponent<T>(this GameObject gameObject) where T : Component
+        {
+            return gameObject.GetComponent<T>() != null;
+        }
+
+        /// <summary>
+        ///     Extension method to check if GameObject has a component.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="gameObject"></param>
+        /// <returns></returns>
+        public static bool HasComponent(this GameObject gameObject, string componentName)
+        {
+            return gameObject.GetComponent(componentName) != null;
+        }
+
+        /// <summary>
+        ///     Check if GameObject has any of the specified components.
+        /// </summary>
+        /// <param name="gameObject"></param>
+        /// <param name="components"></param>
+        /// <returns></returns>
+        public static bool HasAnyComponent(this GameObject gameObject, params Type[] components)
+        {
+            foreach (var compo in components)
+            {
+                if (gameObject.GetComponent(compo) != null)
```

Comment:
```suggestion
                if (gameObject.GetComponent(compo))
```

because of the mentioned null checks. This also applies to the follwing methods.